src/Microsoft.DotNet.Build.CloudTestTasks/RunnerScripts/xunitrunner-func/xunitrunner-func-ilc.py

```diff
@@ -0,0 +1,235 @@
+#!/usr/bin/env py
+import json
+import os.path
+import re
+import shutil
+
+import helix.azure_storage
+import helix.depcheck
+import helix.event
+import helix.logs
+import helix.proc
+import helix.saferequests
+
+from helix.cmdline import command_main
+from helix_test_execution import HelixTestExecution
+from helix.io import ensure_directory_exists
+from helix.settings import settings_from_env
+
+log = helix.logs.get_logger()
+event_client = helix.event.create_from_uri(settings_from_env().event_uri)
+
+
+def _get_xunit_file_list(assembly_list, execution_location):
+    log.info("Opening assembly list from {}".format(assembly_list))
+    log.info("Only Xunit dependencies will be copied as this is a .NET Native compilation run")
+
+    files_and_destinations = []
+    try:
+        assembly_list_obj = json.loads(open(assembly_list).read())
+        try:
+            for assembly_name in assembly_list_obj["xunit"]:
+                assembly_name = assembly_name.replace("/", os.path.sep)
+                assembly_name = assembly_name.replace("\\", os.path.sep)
+                files_and_destinations.append((assembly_name, execution_location))
+            return files_and_destinations
+        except:
+            log.error("Failed parsing " + assembly_list)
+            # this is a fatal error
+            raise
+    except:
+        # Failure to find assembly list
+        raise
+
+
+def _find_file(root, file_name):
+    for (root, _, files) in os.walk(root):
+        if file_name in files:
+            return os.path.join(root, file_name)
+    return None
+
+
+def _create_reflection_directive_file(test_drop, working_source_path):
+        rd_files = [f for f in os.listdir(test_drop) if f.endswith('.rd.xml')]
+        rd_target_path = os.path.join(working_source_path, 'default.rd.xml')
+        ensure_directory_exists(working_source_path)
+        if len(rd_files) != 1:
+            log.info("Creating default.rd.xml")
+            with open(rd_target_path, 'w') as rd_xml:
+                rd_xml.write(
+                    """<Directives xmlns="http://schemas.microsoft.com/netfx/2013/01/metadata">
+                      <Application>
+                        <Assembly Name="*Application*" Dynamic="Required All" />
```

Comment:
Could this be optional? We set up the `default.rd.xml` for UWP apps the way it is mostly because various dependency injection frameworks and reflection-based serializers have unpredictable reflection patterns and fail miserably without this line. "Normal" reflection should just work. It would be nice to be able to run tests just against the built-in compiler reflection heuristics (i.e. without the `*Application*` line in this file).

Not everything will work (?), but I would be interested in the classes of things that don't work.